flake/modules/home-manager/options/misc.nix

```diff
@@ -79,5 +85,43 @@ in {
       // {default = true;};
 
     showHomeButton = mkEnableOption "displaying Home Button in the toolbar";
+
+    customMozillaFolder = {
+      enable = mkEnableOption ''
+        a custom .mozilla folder for storing browser data.
+
+        You may want this so it doesn't clash with other firefox installations.
+      '';
+      path = let
+        # the extra check code has been taken from github.com/nix-community/disko
+        # it was, as far as I can tell written by Lassulus <https://github.com/lassulus>
+        # the project is licensed under MIT License,
+        # which can be found at https://github.com/nix-community/disko/blob/master/LICENSE
+        # Copyright (c) 2018, 2019, 2022–2024 Nix community projects
+        check = x:
+          with lib; let
```

Comment:
Please omit `with lib;` here. I understand that the scope is small, but I am not very keen on using it no matter the context.